crates/db_schema_file/replaceable_schema/triggers.sql

```diff
@@ -576,6 +576,54 @@ CALL r.create_person_saved_combined_trigger ('post');
 
 CALL r.create_person_saved_combined_trigger ('comment');
 
+-- person_liked (comment, post)
+-- This one is a little different, because it triggers using x_actions.liked,
+-- Rather than any row insert
+CREATE PROCEDURE r.create_person_liked_combined_trigger (table_name text)
+LANGUAGE plpgsql
+AS $a$
+BEGIN
+    EXECUTE replace($b$ CREATE FUNCTION r.person_liked_combined_change_values_thing ( )
+            RETURNS TRIGGER
+            LANGUAGE plpgsql
+            AS $$
+            BEGIN
+                IF (TG_OP = 'DELETE') THEN
+                    DELETE FROM person_liked_combined AS p
+                    WHERE p.person_id = OLD.person_id
+                        AND p.thing_id = OLD.thing_id;
+                ELSIF (TG_OP = 'INSERT') THEN
+                    IF NEW.liked IS NOT NULL THEN
+                        INSERT INTO person_liked_combined (liked, like_score, person_id, thing_id)
+                            VALUES (NEW.liked, NEW.like_score, NEW.person_id, NEW.thing_id);
+                    END IF;
+                ELSIF (TG_OP = 'UPDATE') THEN
+                    IF NEW.liked IS NOT NULL THEN
+                        INSERT INTO person_liked_combined (liked, like_score, person_id, thing_id)
+                            VALUES (NEW.liked, NEW.like_score, NEW.person_id, NEW.thing_id);
+                        -- If liked gets set as null, delete the row
+                    ELSE
+                        DELETE FROM person_liked_combined AS p
+                        WHERE p.person_id = NEW.person_id
+                            AND p.thing_id = NEW.thing_id;
+                    END IF;
+                END IF;
+                RETURN NULL;
+            END $$;
+    CREATE TRIGGER person_liked_combined
+        AFTER INSERT OR DELETE OR UPDATE OF liked ON thing_actions
+        FOR EACH ROW
+        EXECUTE FUNCTION r.person_liked_combined_change_values_thing ( );
+    $b$,
+    'thing',
+    table_name);
+END;
+$a$;
```

Comment:
Should only insert rows if the vote was done by a local user. In fact it can be changed from person_id to local_user_id.

Reply:
This is a tough one.

The other similar combined views are 

- `person_content_combined` (which does need to be shown for federated users)
- `person_saved_combined` (which like this one, could be changed to local_user, since it isn't public)`.

There's also the fact that `post_actions` and `comment_actions` use the `person_id`, when in reality most of those fields aren't public, and could be made to `local_user` anyway.

I think for now, since `post_actions / comment_actions` are attached to person, we should leave these attached to the person, until those are reworked.

cc @dullbananas

Reply:
Its fine to use person as key, but no need to store all this data for remote users.

Reply:
Hrm... I'm realizing this is tougher than it seems. 

@dullbananas `person_saved` for example, is another post action that really only needs to apply to local users, not federated ones. We could either add some join check [here](https://github.com/LemmyNet/lemmy/blob/main/crates/db_schema_file/replaceable_schema/triggers.sql#L549), or rework the actions / table triggers. What do you think?

Reply:
@Nutomic is right. The triggers should not insert votes from remote users, but just because a column can reference local user instead of person doesn't mean it should. Use person id.

Reply:
My question is, how do we enforce that these triggers are only altering data for local users only?

- For `person_saved_combined`, that action doesn't send out any federated action, so we're fine there by default.
- For this PR, `person_liked_combined`, we **are** sending and receiving likes for federated users, but the `person_liked_combined` table should only be holding data for local users.

Reply:
You could add post_actions.local and comment_actions.local columns, and check those in the trigger. It's also possible to check `(select local from person where id = NEW.person_id)` to avoid adding a redundant column, but it makes the trigger slower, and a redundant "local" column is what we currently do for other tables.

Reply:
That does seem like the best solution, thx.

Reply:
For now just use `(select local from person where id = NEW.person_id)`. Also may or may not need to check `OLD.person_id`. I will open a separate issue for adding the column to the actions tables.

Reply:
Done, added this for the inserts to liked / saved combined.